apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -556,3 +572,7 @@ func (kc *K8ssandraCluster) DefaultNumTokens(serverVersion *semver.Version) floa
 func (kc *K8ssandraCluster) GetClusterIdHash() string {
 	return utils.HashNameNamespace(kc.Name, kc.Namespace)
 }
+
+func (k *K8ssandraCluster) GenerationChanged() bool {
+	return k.Status.ObservedGeneration < k.Generation
```

Comment:
[thought] Since this PR also introduces the new field, maybe we should add this to correctly detect the first operator upgrade:
```suggestion
	return k.Status.ObservedGeneration != 0 && k.Status.ObservedGeneration < k.Generation
```

Reply:
Turns out, this causes a bug. ObservedGeneration is 0 in new clusters also and that's perfectly valid - so we have to allow update in that case, otherwise we'll never update the CassandraDatacenters. I think we'll just have to accept that older clusters won't benefit from restart protection the first time.